### ISSUE TITLE
Soda using Trino having issue on the authentication for HTTP

### ISSUE THREAD

**Issue by Giisela** [sodadata/soda-core#1653]
I'm trying to use soda with trino and I am facing an issue that I'm not sure if I'm doing something wrong here:

I'm using docker with a trino image, the version is 397. And didn't set up any password since I was testing locally (using HTTP).
Using trino-python-cliente with DBAPI connection (https://github.com/trinodb/trino-python-client) I was able to connect to my local trino without the use of a password, only passing a username, host, port and catalog.

Using soda scan, I set the configuration.yml as follow :
```
data_source test:
  type: trino
  host: localhost
  port: 8080
  username: trino
  password: *****
  catalog: iceberg
  schema: default
```
When I run soda scan the output was:
```
HTTPSConnectionPool(host='localhost', port=8080): Max retries exceeded with url: /v1/statement (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:997)')))
Error occurred while executing scan.
  | no transaction was started
```
Also I tried to remove the field but I saw on the documentation that it is a required field (https://docs.soda.io/soda/connect-trino.html).

I saw that on the source code we were able to pass a `http_scheme` that by default it was set to https, but I set it to http and didn't work (https://github.com/sodadata/soda-core/blob/main/soda/trino/soda/data_sources/trino_data_source.py#L79).

I notice this line (https://github.com/sodadata/soda-core/blob/main/soda/trino/soda/data_sources/trino_data_source.py#L80) that forces the Basic Auth authentication, and I'm afraid that my problem can be because of this line since I'm trying to use HTTP that doesn't require any password.
Could we have the option where whenever we want to use HTTP, the basic auth authentication isn't required and we can pass only the user instead of the user and password? That could help with local tests.

**Comment by jmarien** [comment]
SODA-1250

**Comment by vijaykiran** [comment]
Thanks for the issue @Giisela  I'll make the password optional and that should probably resolve this.

**Comment by ScottAtDisney** [comment]
[PR #1658](https://github.com/sodadata/soda-core/pull/1658) was submitted to resolve this.

**Comment by vijaykiran** [comment]
Fixed in 3.0.12   @Giisela  Can you please try and let us know?

**Comment by pcriadoperez** [comment]
It worked!
Here I'll share my configuration if it helps anyone else:
```
data_source trino:
  type: trino
  host: 127.0.0.1
  port: "8080"
  username: admin
  auth_type: "NoAuthentication" 
  http_scheme: http
```
